### DISCUSSION TITLE
Can't invoke onNameLookup event

### DISCUSSION THREAD

**Discussion by hemulin** [MetaMask/snaps#2892] [Q&A] [answered]
Hi there,

I'm building a snap that is intended to resolve domain names.

After reading documentation and looking at some templates and examples, I have a manifest that contains:
```
"initialPermissions": {
    "endowment:ethereum-provider": {},
    "endowment:network-access": {},
    "endowment:name-lookup": {
      "chains": [
        "eip155:1",
        "eip155:137"
      ]
    }
  }
```

And a method with signature:
```
export const onNameLookup: OnNameLookupHandler = async (request) => {...}
```

However, when I access the extension UI => Send => To, and typing a domain, the method is never being invoked.

I do see in the console an indication that another snap (`@metamask/ens-resolver-snap`) is attempting to resolve the domain
(below is output while me typing in the input field)
```
entry-install.js:1 RPC (npm:@metamask/ens-resolver-snap): metamask_getProviderState -> success
sentry-install.js:1 RPC (npm:@metamask/ens-resolver-snap): eth_chainId -> success
sentry-install.js:1 RPC (npm:@metamask/ens-resolver-snap): eth_chainId -> success
sentry-install.js:1 RPC (npm:@metamask/ens-resolver-snap): eth_call -> success
sentry-install.js:1 RPC (npm:@metamask/ens-resolver-snap): eth_chainId -> success
sentry-install.js:1 RPC (npm:@metamask/ens-resolver-snap): eth_call -> success
sentry-install.js:1 RPC (npm:@metamask/ens-resolver-snap): eth_chainId -> success
sentry-install.js:1 RPC (npm:@metamask/ens-resolver-snap): eth_call -> success
```

I haven't installed the ens-resolver-snap, so I assume it is shipped out of the box with the extension.

Am I missing something? Can it be that this snap hijacks the event I'm trying to capture? Any idea how to fix it and get my method triggered?

I'm running Metamask flask version 12.6.0-flask.0

Thanks in advance.

**Comment by Montoya** [comment]
Are you sure you are on ETH Mainnet or Polygon Mainnet when you are trying the send flow? Based on your manifest, your Snap will only be called on those two networks. 

Also, it's possible that your Snap has a bug that is causing the resolution to fail. Can you share the contents of the handler? 

`export const onNameLookup: OnNameLookupHandler = async (request) => {...}`

**Comment by hemulin** [comment]
Hi @Montoya,

Yes, I'm certain that this runs on Mainnet or Polygon when trying.

Here is the handler:
```
export const onNameLookup: OnNameLookupHandler = async (request) => {
  console.log('#1: onNameLookup', request)
  const { chainId, domain } = request

  const baseURL = process.env.RESOLVING_API_URL;
  const supportedChains = ["eip155:1", "eip155:137"];
  const supportedRegistries = ["FNS", "ENS", "UD"];

  if (domain && supportedChains.includes(chainId)) {
    for (const registry of supportedRegistries) {
      const response = await fetch(`${baseURL}/${registry}/${domain}`)
      const data = await response.json()
      console.log('#2: onNameLookup', data)
      const resolvedAddress = data.address
      if (resolvedAddress) {
        return {
          resolvedAddresses: [
            { resolvedAddress, protocol: registry, domainName: domain },
          ],
        }
      }
    }
  }

  console.log('#3: onNameLookup - null')
  return null
};
```

None of the debug prints is presented in the console

**Comment by hemulin** [comment]
Greetings @Montoya 

Do you see anything wrong in my handler? I feel like I'm missing something fundamental but I have no idea what it might be.

I'm using the snap template and everything seems to be on par with the documentation.

**Reply by hmalik88** [reply]
Can you post the logs that you're seeing when testing? What are your test values? If nothing in the handler is being called, to make sure, is the snap installed correctly?

**Reply by hemulin** [reply]
I've uninstalled the MM flask extension, installed it again, ran the snap template code, connected and installed the snap, validated it is installed in the snaps menu (the only snap installed).

Please see attached screen record of the attempt to invoke the `onNameLookup`.

None of the console.log that are in the handler are printed.

https://github.com/user-attachments/assets/6c47efc7-ec8a-4020-ad16-4dfc50b3caf9

**Reply by hemulin** [reply, marked as answer]
Sorry for the false alarm. Everything is working as expected.
I tried debugging with console.log and apparently this is being swallowed somewhere along the way (nothing I could find in the flask extension console, nor in the browser).

Anyhow, simply returning hardcoded:
```
return {
    resolvedAddresses: [
      {
        resolvedAddress: '0x0',
        protocol: "${valueToObserve1}",
        domainName: "${valueToObserve2}"
      }
    ]
  };
```

Was helpful enough to get me going and figure out each step along the way.

(and many thanks to [alexander2001i](https://github.com/alexander2001i) for his assistance)
